### DISCUSSION TITLE
GraphQL PriceAdjustment Object Additions

### DISCUSSION THREAD

**Discussion by devinatbryt** [Shopify/function-examples#217] [Feedback: Cart Transform API]
## Overview
*I'm working on an app integration that adds items to a bundle dynamically based off of a few specific parameters that ties them to the individual bundle.*

### Request
 It would be nice be able to adjust the price based off of the parent bundle product's price, instead of only getting access to a percentage based adjustment.
 
 ### Possible Integrations?
 1. Similar to the discount api, the price adjustment object should be structured as so:
 ```
 {
    price {
        fixedDecrease {
            amount: $amount
        }
    }
 }
 ```
 and then pair the following with adding some price field on the actual ProductVariant option itself, so you'd be able to calculate the difference between the current item totals and the parent bundle product's price.
 
 2. Maybe a little different but instead have the calculations happen on Shopify's end and have the graphql PriceAdjustment look like the following?
 ```
 {
     price {
         compareAtPrice: true
     }
 }
 ```
 
 ### Final Words/Questions
 I'm still entirely unsure of the naming conventions, but I think it would be nice to have a little bit more information about the products and it would certainly be nice to set the merged items to the parent bundle price to allow the ability to change the price of a bundle, based off of totals defined by the "client" instead of a set percentage. Thanks for taking the time to read over my feedback/request! :)

**Comment by ignacio-chiazzo** [comment]
Hey, thanks for the feedback. 

I'm wondering if something like this issue would be helpful https://github.com/Shopify/function-examples/discussions/225. Could you elaborate on your use case(s)?

**Reply by bureauvk** [reply]
If you want me to talk over a couple of use cases with you, feel free to send me a message.
